### PR TITLE
New Stubbing arguments

### LINKED ISSUE #137: Multiple stub-related parameters are redundant
Having `stubResponses` and `stubBehavior` seems redundant. We should infer stub behaviour based on the presence of an optional `stubBehavior`.

### LINKED ISSUE #142: Clean up default args in initializer
Some are references to functions, some are invocations of functions that return closures. It'd be nice to be consistent.

### PR DESCRIPTION
**Pull request by ashfurrow**
Removes `stubResponses` in favour of a new `StubbingBehavior` (it would be weird before if you said "don't stub responses", but also "delay my stubbed responses"). 

Also redefines types of `DefaultEndpointMapping` and `DefaultEnpointResolution` class functions so you don't invoke them to get the direct values anymore.

Old:

```swift
MoyaProvider<MyTarget>(endpointResolver: MoyaProvider.DefaultEndpointResolution())
```

vs. new:

```swift
MoyaProvider<MyTarget>(endpointResolver: MoyaProvider.DefaultEndpointResolution)
```

Fixes #137.
Fixes #142.

### CODE REVIEW

**Comment by ashfurrow** [comment]
Some pretty far-reaching changes here. Would appreciate a review.

File: Demo/DemoTests/MoyaProviderSpec.swift [hunk 1]
```diff
@@ -143,7 +139,7 @@ class MoyaProviderSpec: QuickSpec {
                             executed = true
                             return endpoint.urlRequest
                         }
-                        provider = MoyaProvider(endpointsClosure: endpointsClosure, endpointResolver: endpointResolution, stubResponses: true)
+                        provider = MoyaProvider<GitHub>(endpointResolver: endpointResolution, stubBehavior: MoyaProvider.ImmediateStubbingBehaviour)
```
Comment (colinta) [inline Demo/DemoTests/MoyaProviderSpec.swift]: `endpointsClosure` vs `endpointResolver` caught my eye ("endpoints" vs "endpoint"), are these "similar-yet-different"?  I'm not familiar w/ the resolver.

Comment (ashfurrow) [inline Demo/DemoTests/MoyaProviderSpec.swift]: Excellent catch. Will update.

**Comment by alcarvalho** [comment]
Everything looks fine to me. The new code is clearer, indeed. Nice job! ;)

**Pull request merged**